find/main/routes.py

```diff
@@ -175,6 +178,49 @@ def retrieve_download(filename: str):
         return render_template("find/main/retrieve-download.html", context=context, form=form)
 
 
+@bp.route("/retrieve-spreadsheet/<fund_code>/<submission_id>", methods=["GET", "POST"])
+@login_required(return_app=SupportedApp.POST_AWARD_FRONTEND)
+@check_internal_user
+def retrieve_spreadsheet(fund_code: str, submission_id: str):
+    object_name = f"{fund_code}/{submission_id}"
+
+    try:
+        file_metadata = get_file_metadata(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES, object_name, True)
+    except FileNotFoundError:
+        if request.method == "POST":
+            return redirect(
+                url_for(
+                    ".retrieve_spreadsheet",
+                    fund_code=fund_code,
+                    submission_id=submission_id,
+                )
+            )
+
+        return render_template("find/main/spreadsheet-not-found.html")
+
+    filename = f"{fund_code}---{submission_id}.xlsx"
+    file_size = round(file_metadata["ContentLength"] / (1024 * 1024), 1)
```

Comment:
We have a function called `get_human_readable_file_size` in `async_download` - we can probably move this to some shared/central place and use that instead?

Reply:
There's also `get_find_download_file_metadata` which does some of what you've added with the `return_full_head_data` - can we come up with something consistent and standardise on that?

Reply:
P.S. I'm happy for refactors to be a separate PR if it helps keep things small/contained/reviewable 🙏